jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/BusinessAdminPreprocessorTest.java

```diff
@@ -0,0 +1,97 @@
+package org.jbpm.kie.services.impl.query.preprocessor;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.dashbuilder.dataset.DataSetLookup;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.internal.identity.IdentityProvider;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BusinessAdminPreprocessorTest {
+
+    @Mock
+    IdentityProvider identityProvider;
+
+    DataSetLookup dataSetLookup;
+
+    //@InjectMocks
+    BusinessAdminPreprocessor businessAdminPreprocessor;
+
+    @Before
+    public void init() {
+        System.setProperty("org.jbpm.ht.admin.group","admins");
+        System.setProperty("org.jbpm.ht.admin.user","admin");
+        businessAdminPreprocessor= new BusinessAdminPreprocessor(identityProvider);
+        dataSetLookup = new DataSetLookup();
+    }
+
+    @Test
+    public void testQueryByBusinessAdmin() {
+        String role1 = "admins";
+        String role2 = "role2";
+        String userId = "userId";
+
+        when(identityProvider.getRoles()).thenReturn(Arrays.asList(role1, role2));
+        when(identityProvider.getName()).thenReturn(userId);
+
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+        assertNull(dataSetLookup.getFirstFilterOp());
+
+        role1 = "role1";
+        role2 = "role2";
+        userId = "admin";
+
+        when(identityProvider.getRoles()).thenReturn(Arrays.asList(role1, role2));
+        when(identityProvider.getName()).thenReturn(userId);
+
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+        assertNull(dataSetLookup.getFirstFilterOp());
+    }
+
+    @Test
+    public void testQueryByNoBusinessAdmin() {
+        String role1 = "role1";
+        String role2 = "role2";
+        String userId = "userId";
+
+        when(identityProvider.getRoles()).thenReturn(Arrays.asList(role1, role2));
+        when(identityProvider.getName()).thenReturn(userId);
+
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+        assertEquals("TASKID = -1", dataSetLookup.getFirstFilterOp().getColumnFilterList().get(0).toString());
+
+        when(identityProvider.getRoles()).thenReturn(Collections.emptyList());
+        when(identityProvider.getName()).thenReturn(userId);
+
+        dataSetLookup= new DataSetLookup();
+        businessAdminPreprocessor.preprocess(dataSetLookup);
+
+
+        assertEquals("TASKID = -1", dataSetLookup.getFirstFilterOp().getColumnFilterList().get(0).toString());
+    }
+
+    @Test
+    public void testSetUserWithoutRoles() {
```

Comment:
This test is a duplicate of the previous test (lines 75-82).
However, I'd encourage you to leave this test as it is and remove lines 75-82 for clarity reasons.